src/adera/Machines/Rocket.cpp

```diff
@@ -163,9 +170,86 @@ void SysMachineRocket::update_physics()
     }
 }
 
+void SysMachineRocket::attach_plume_effect(ActiveEnt ent)
+{
+    using Magnum::GL::Mesh;
+    using Magnum::GL::Texture2D;
+    using Magnum::Trade::ImageData2D;
+    using namespace Magnum::Math::Literals;
+
+    ActiveEnt plumeNode = entt::null;
+
+    auto findPlumeHandle = [&](ActiveEnt ent)
+    {
+        ACompHierarchy& node = m_scene.reg_get<ACompHierarchy>(ent);
+        if (node.m_name.compare(0, 9, "fx_plume_") == 0)
+        {
+            plumeNode = ent;
+            return false;  // terminate search
+        }
+        return true;
+    };
+
+    m_scene.hierarchy_traverse(ent, findPlumeHandle);
+
+    if (plumeNode == entt::null)
+    {
+        std::cout << "ERROR: could not find plume anchor for MachineRocket "
+            << static_cast<int>(ent) << "\n";
+        return;
+    }
+    std::cout << "MachineRocket "
+        << static_cast<int>(ent) << "'s associated plume: "
+        << static_cast<int>(plumeNode) << "\n";
+
+    // Get plume effect
+    Package& pkg = m_scene.get_application().debug_find_package("lzdb");
+    std::string plumeAnchorName = m_scene.reg_get<ACompHierarchy>(plumeNode).m_name;
```

Comment:
The underlying value from reg_get<...>(...) is stable in memory right? If so, take the plumeAchorName as either a std::string const&, or a std::string_view.

Reply:
Just fixed this moments ago in a yet-unpushed commit. Since the name is just used to query the resources, they can all be `std::string_view`